### DISCUSSION TITLE
GitHub Classroom Autograding Not Removing Newlines

### DISCUSSION THREAD

**Discussion by dragoonius** [community/community#64984] [GitHub Education]
### Hi everyone,

Hello. I seem to be having an odd problem when including multiple inputs in my tests. For some reason the autograder is including the newline at the end of the line in the string that it is storing in a variable,  For example, the program asks the user for their name and age, and the output should be:

Please enter your name: Please enter your age: Hello, Bob you are 35 years old!

But instead the output is:

Please enter your name: Please enter your age: Hello, Bob
 you are 35 years old!

When I look at the error message it shows me:

❌ Hello3 Test
::error::The output for test Hello3 Test did not match%0AExpected:%0AHello, Bob you are 35 years old!%0AActual:%0APlease enter your name: Please enter your age: Hello, Bob%0D you are 35 years old!

::***::

I'm just not sure why it is including the %0D, i.e. the newline, in the first input string. Any help would be greatly appreciated.

**Comment by arafajs** [comment]
The %0D character is a newline character in hexadecimal. It is possible that the autograder is interpreting the newline character in the first input string as a hexadecimal character. To fix this, you can use the r prefix to tell the autograder to interpret the string as a raw string.

**Reply by dragoonius** [reply]
That sounds like it could work! I'm not quite sure how to use the r prefix though, as this is my first time using the autograder. Where would I add the r prefix?

**Reply by jzaldivar** [reply]
I think he was referring that, for the output string, use: "Hello, Bob\r\n you are 35 years old!"

The "\r\n" sequence is the equivalent of a carriage-return + new-line.

You can try just "\r" or just "\n" or the combination "\r\n", to check which one works for you. I, myself, use "\r\n" in the input string and just "\n" in the output.
